### PR TITLE
Closes issue 54

### LINKED ISSUE #54: Antlr is not parsing specific numbers correctly
This unit test fails unexpectedly:

```c#        
[Fact]
public void IfTest()
{
    // Arrange
    const long expected = 9999999999L;
    var expression = $"if(true, {expected}, 0)";
    var e = new Expression(expression);

    // Act
    var actual = e.Evaluate();

    // Assert
    Assert.Equal(expected, actual);
}
```

> Xunit.Sdk.EqualException
Assert.Equal() Failure
Expected: 9999999999
Actual:   1E+10

Strangely the test passes if you remove one `9` from the `long` constant. The same thing happens for the lambda version:

```c#
[Fact]
public void IfTest2()
{
    // Arrange
    const long expected = 9999999999L;
    var expression = $"if(true, {expected}, 0)";
    var e = new Expression(expression);
    var context = new object();

    var lambda = e.ToLambda<object, long>();

    // Act
    var actual = lambda(context);

    // Assert
    Assert.Equal(expected, actual);
}
```

### PR DESCRIPTION
**Pull request by rexcfnghk**
Closes #54

### CODE REVIEW

File: src/NCalc/NCalcParser.cs [hunk 1]
```diff
@@ -2587,7 +2587,7 @@ private NCalcParser.value_return value()
 				adaptor.AddChild(root_0, INTEGER35_tree);
 
 				DebugLocation(208, 14);
-				 try { retval.value = new ValueExpression(int.Parse((INTEGER35!=null?INTEGER35.Text:null))); } catch(System.OverflowException) { retval.value = new ValueExpression(long.Parse((INTEGER35!=null?INTEGER35.Text:null))); } 
+				 try { retval.value = new ValueExpression(int.Parse((INTEGER35!=null?INTEGER35.Text:null))); } catch(System.OverflowException) { retval.value = new ValueExpression((object)long.Parse((INTEGER35!=null?INTEGER35.Text:null))); }
```
Comment (sklose) [inline src/NCalc/NCalcParser.cs:2590]: is there a way to change this in the grammar file itself? this file is generated code and ideally shouldn't be changed manually

Comment (rexcfnghk) [reply src/NCalc/NCalcParser.cs:2590]: Sorry I am not sure how. Which file should I look into?

Comment (Tradioyes) [reply src/NCalc/NCalcParser.cs:2590]: I think it's here: https://github.com/sklose/NCalc2/blob/e9f2d1b73f2e680a01c66eed1ade5958b43524b5/grammer/NCalc.g#L209

Comment (rexcfnghk) [reply src/NCalc/NCalcParser.cs:2590]: Thanks @Tradioyes. Added the change

**Pull request merged**